### PR TITLE
Added locale loading, polyfill for Intl for Safari, user config file,…

### PR DESCRIPTION
**Pull request by benjaminlgt-littlelives**
… support for polling for webpack, extraction of messages and message loading.

### CODE REVIEW

**Comment by SilentCicero** [comment]
Looking good ! Lets have Max look it over @benjaminlgt-littlelives

**Comment by mxstbr** [comment]
Awesome, thanks so much @benjaminlgt-littlelives!

I'm not a big fan of the `app/config.json` file. Why do we need a special Babel config in there?

Instead, we should just move that to the general Babel config, and move the locales to the `app/bootloader.js` file. Also, what about then renaming that from `bootloader.js` to `i18n.js` or something like that?

**Comment by mxstbr** [comment]
Also, what @justingreenberg mentioned over here https://github.com/mxstbr/react-boilerplate/pull/391#issuecomment-221916649, we shouldn't be adding the Intl polyfill as a script tag, that's a brilliant point!

**Comment by SilentCicero** [comment]
I'm in favour of `i18n.js` or `intl.js` @mxstbr @benjaminlgt-littlelives

**Comment by mxstbr** [comment]
Alright, lets do `i18n.js`!

**Comment by benjaminlgt-littlelives** [comment]
@mxstbr Sounds good, adopted the changes as discussed.
@justingreenberg's polyfill method is done as well.

**Comment by mxstbr** [comment]
Two more things:
1. Merge `i18n.js` and `bootloader.js` and call the file `i18n.js`. There's no reason to have them separate as far as I can see. :blush:
2. Add comments to the `bootloader.js` part of the code, explaining what everything is used for, and the same thing for `internals/scripts/extractIntl.js` because I have no idea what that does and how it works tbh. 🙈 

Looks good otherwise, thanks @benjaminlgt-littlelives!

File: app/app.js [hunk 1]
```diff
@@ -59,9 +62,9 @@ const rootRoute = {
   childRoutes: createRoutes(store),
 };
 
-ReactDOM.render(
+const render = () => ReactDOM.render(
   <Provider store={store}>
-    <IntlProvider locale="en">
+    <IntlProvider locale="de" messages={translationMessages.en}>
```
Comment (justingreenberg) [inline app/app.js:67]: i think locale prop should be 'en' here

File: app/app.js [hunk 2]
```diff
@@ -90,6 +93,16 @@ ReactDOM.render(
   document.getElementById('app')
 );
 
+// Chunked polyfill for browsers without Intl support
+if (!window.Intl) {
+  require.ensure(['intl'], (require) => {
```
Comment (justingreenberg) [inline app/app.js:98]: any reason not using system import?

File: internals/webpack/webpack.base.babel.js [hunk 3]
```diff
@@ -88,4 +88,9 @@ module.exports = (options) => ({
   target: 'web', // Make web variables accessible to webpack, e.g. window
   stats: false, // Don't show stats in the console
   progress: true,
+
+  // Polling for vagrant based development
```
Comment (justingreenberg) [inline internals/webpack/webpack.base.babel.js:92]: since polling is not necessary for i18n and is CPU-intensive, maybe this should be moved into separate pr... i don't use vagrant/docker, but i understand this is an important feature for those environments, so maybe this could be another PR but moving the poll value into config but leaving default as false? @mxstbr thoughts?

File: internals/webpack/webpack.base.babel.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ module.exports = (options) => ({
   }, options.output), // Merge with env dependent settings
   module: {
     loaders: [{
-      test: /\.js$/, // Transform all .js files required somewhere with Babel
+      test: /\.js(on)?$/, // Transform all .js, .json files required somewhere with Babel
```
Comment (justingreenberg) [inline internals/webpack/webpack.base.babel.js:16]: again, not to be picky at all but json shouldn't need to be transpiled, we should use json-loader

**Comment by justingreenberg** [comment]
@benjaminlgt-littlelives left a few little comments—awesome work dude

File: internals/webpack/webpack.base.babel.js (hunk 3 above)
Comment (mxstbr) [inline internals/webpack/webpack.base.babel.js:92]: Yeah I agree, was going to mention that but forgot. This should definitely not be included by default!

**Comment by SilentCicero** [comment]
@mxstbr @benjaminlgt-littlelives @justingreenberg are we good to merge this, then I'll make the changes to staging a `messages.js` file for each component?

**Comment by mxstbr** [comment]
The few little comments by @justingreenberg should be adapted, then we can merge this and add the messages.js files! :+1:

File: app/app.js (hunk 1 above)
Comment (benjaminlgt-littlelives) [inline app/app.js:67]: Oops, yes you're right.

File: internals/webpack/webpack.base.babel.js (hunk 3 above)
Comment (benjaminlgt-littlelives) [inline internals/webpack/webpack.base.babel.js:92]: Noted, will remove this.
